compiler/main/checks.cpp

```diff
@@ -622,6 +624,16 @@ checkResolveRemovedPrims(void) {
   }
 }
 
+static void checkNoRecordDeletes() {
+  // No need to do for_alive_in_Vec - there shouldn't be any, period.
+  // User errors are to be detected by chpl__delete() in the modules.
+  forv_Vec(CallExpr, call, gCallExprs)
+    if (FnSymbol* fn = call->resolvedFunction())
+      if(fn->hasFlag(FLAG_DESTRUCTOR))
+        if (isRecord(call->get(1)->typeInfo()->getValType()))
+          INT_FATAL(call, "delete on records");
```

Comment:
I assumed this is a separate compiler error, and that we could write a test to demonstrate that as well - did I assume incorrectly?